packages/site/cms-site/src/iframebridge/useBlockPreviewFetch.tsx

```diff
@@ -16,16 +16,18 @@ export function useBlockPreviewFetch(apiUrl: string): { fetch: Fetch; graphQLFet
 export function useBlockPreviewFetch(apiUrl?: string | undefined): { fetch: Fetch; graphQLFetch?: GraphQLFetch };
 export function useBlockPreviewFetch(apiUrl?: string | undefined) {
     const { showOnlyVisible, graphQLApiUrl } = useIFrameBridge();
+    const [graphQLFetch, setGraphQLFetch] = useState<GraphQLFetch | undefined>(() =>
+        apiUrl ? createBlockPreviewFetch(apiUrl, !showOnlyVisible) : undefined,
+    );
 
-    const graphQLFetchRef = useRef(apiUrl ? createBlockPreviewFetch(apiUrl, !showOnlyVisible) : undefined);
     useEffect(() => {
         if (graphQLApiUrl) {
-            graphQLFetchRef.current = createBlockPreviewFetch(graphQLApiUrl, !showOnlyVisible);
+            setGraphQLFetch(() => createBlockPreviewFetch(graphQLApiUrl, !showOnlyVisible));
```

Comment:
```suggestion
            setGraphQLFetch(createBlockPreviewFetch(graphQLApiUrl, !showOnlyVisible));
```

No need for an updater function here.

Reply:
The fetch function would be executed and not stored without the update function.

Reply:
Let's add a comment vor this:

```suggestion
            // Use updater function because the function returned by createBlockPreviewFetch would be treated as updater function otherwise.
            setGraphQLFetch(() => createBlockPreviewFetch(graphQLApiUrl, !showOnlyVisible));
```